client/src/components/web-extension/Staking/StakingGrid/RewardInfo.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getStakingRewards } from '@cd/actions/userActions';
+import { userStakingRewardSelector } from '@cd/selectors/user';
+import { MiddleTruncatedText } from '../../../Common/MiddleTruncatedText';
+import Grid from '../../Common/Grid';
+import './StakingInfo.scss';
+
+const REWARDS_METADATA = {
+	left: [
+		{ key: 'validatorPublicKey', type: 'primary', wrapperComponent: MiddleTruncatedText },
+		{ key: 'era', type: 'secondary', format: 'number' },
+	],
+	right: [
+		{ key: 'amount', type: 'primary', format: 'mote', suffix: 'CSPR' },
+		{
+			key: 'createdAt',
+			type: 'secondary',
+			format: 'date',
+			formatOptions: { timeStyle: 'short', dateStyle: 'short' },
+		},
+	],
+};
+
+export const RewardInfo = ({ publicKey }) => {
+	const dispatch = useDispatch();
+	const { isLoading } = useSelector(userStakingRewardSelector);
+	const [rewardPage, setRewardPage] = useState(1);
+	const [rewardData, setRewardData] = useState({});
+
+	useEffect(() => {
+		const loadReward = async () => {
+			const { data } = await dispatch(getStakingRewards(publicKey, 1));
+			setRewardData(data);
```

Comment:
Minor: this should check for equality before setting state to prevent rerender.
```
if (data && !isEqual(rewardData, data)) {
  setRewardData(data);
}
```

Reply:
I think no need here, if we compare with rewardData, need to add rewardData as deps, then it's will trigger useEffect one more time on setRewardData